### ISSUE TITLE
Could Anbox be preventing suspend?

### ISSUE THREAD

**Issue by DroWnThePoor** [anbox/anbox#393]
Make sure you are running the latest version of Anbox before reporting an issue.

Please also check that no similar bug is already reported. Have a look on the list of open bugs at https://github.com/anbox/anbox/issues

** Please paste the result of `anbox system-info` below:**
```
snap-revision: 44
os:
  name: elementary OS
  version: 0.4.1 Loki
  snap-based: true
kernel:
  version: Linux version 4.4.0-87-generic (buildd@lcy01-31) (gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4) ) #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017
  binder: true
  ashmem: true
graphics:
  egl:
    vendor: Mesa Project
    version: 1.5 (DRI2)
    extensions:
      - EGL_CHROMIUM_sync_control
      - EGL_EXT_create_context_robustness
      - EGL_EXT_image_dma_buf_import
      - EGL_KHR_cl_event2
      - EGL_KHR_create_context
      - EGL_KHR_fence_sync
      - EGL_KHR_get_all_proc_addresses
      - EGL_KHR_gl_colorspace
      - EGL_KHR_gl_renderbuffer_image
      - EGL_KHR_gl_texture_2D_image
      - EGL_KHR_gl_texture_3D_image
      - EGL_KHR_gl_texture_cubemap_image
      - EGL_KHR_image
      - EGL_KHR_image_base
      - EGL_KHR_image_pixmap
      - EGL_KHR_reusable_sync
      - EGL_KHR_surfaceless_context
      - EGL_KHR_wait_sync
      - EGL_MESA_configless_context
      - EGL_MESA_drm_image
      - EGL_MESA_image_dma_buf_export
      - EGL_NOK_swap_region
      - EGL_NOK_texture_from_pixmap
      - EGL_NV_post_sub_buffer
      - EGL_WL_bind_wayland_display
  gles2:
    vendor: X.Org
    vendor: OpenGL ES-CM 1.1 Mesa 12.0.6
    extensions:
      - GL_EXT_blend_minmax
      - GL_EXT_multi_draw_arrays
      - GL_EXT_texture_filter_anisotropic
      - GL_EXT_texture_lod_bias
      - GL_OES_byte_coordinates
      - GL_OES_fixed_point
      - GL_OES_stencil_wrap
      - GL_OES_compressed_paletted_texture
      - GL_OES_query_matrix
      - GL_OES_read_format
      - GL_OES_single_precision
      - GL_OES_draw_texture
      - GL_OES_point_size_array
      - GL_OES_point_sprite
      - GL_EXT_texture_format_BGRA8888
      - GL_OES_compressed_ETC1_RGB8_texture
      - GL_OES_depth24
      - GL_OES_element_index_uint
      - GL_OES_fbo_render_mipmap
      - GL_OES_framebuffer_object
      - GL_OES_mapbuffer
      - GL_OES_rgb8_rgba8
      - GL_OES_stencil8
      - GL_OES_texture_env_crossbar
      - GL_OES_texture_mirrored_repeat
      - GL_OES_texture_npot
      - GL_OES_EGL_image
      - GL_OES_packed_depth_stencil
      - GL_OES_texture_cube_map
      - GL_APPLE_texture_max_level
      - GL_EXT_discard_framebuffer
      - GL_EXT_read_format_bgra
      - GL_OES_blend_equation_separate
      - GL_OES_blend_func_separate
      - GL_OES_blend_subtract
      - GL_OES_EGL_image_external
      - GL_OES_EGL_sync
      - GL_OES_vertex_array_object
      - GL_EXT_map_buffer_range
      - GL_KHR_debug
      - GL_OES_surfaceless_context
      - GL_EXT_polygon_offset_clamp

[please paste printout of `anbox system-info` here]
```

**Please describe your problem:**

Computer immediately wakes from sleep after suspending.

Not sure if Anbox is the problem, but I'm wondering if it's a possibility?
Never had this issue before installing/running Anbox.

**Comment by morphis** [comment]
> Not sure if Anbox is the problem, but I'm wondering if it's a possibility?

That shouldn't be the case. Does the problem disappear when you unload either the `binder_linux` or `ashmem_linux` kernel modules or when you remove anbox entirely from your system?

**Comment by Null-Senpai** [comment]
I also have this issue. For me `systemctl --user stop anbox-session-manager.service` made it work again.
That said, I don't think I have the recommended/standard setup (I'm on an Arch based distro, so afaik no snap stuff was used.)

`anbox system-info`:

version: local-14f791a
os:
  name: Antergos Linux
  version: 17.6-ISO-Rolling
  snap-based: false
kernel:
  version: Linux version 4.11.12-1-ck-ivybridge (squishy@ease) (gcc version 7.1.1 20170630 (GCC) ) #1 SMP PREEMPT Fri Jul 21 16:43:37 EDT 2017
  binder: true
  ashmem: true
graphics:
  egl:
    vendor: Mesa Project
    version: 1.4 (DRI2)
    extensions:
      - EGL_CHROMIUM_sync_control
      - EGL_EXT_buffer_age
      - EGL_EXT_create_context_robustness
      - EGL_EXT_image_dma_buf_import
      - EGL_KHR_config_attribs
      - EGL_KHR_create_context
      - EGL_KHR_fence_sync
      - EGL_KHR_get_all_proc_addresses
      - EGL_KHR_gl_renderbuffer_image
      - EGL_KHR_gl_texture_2D_image
      - EGL_KHR_gl_texture_3D_image
      - EGL_KHR_gl_texture_cubemap_image
      - EGL_KHR_image
      - EGL_KHR_image_base
      - EGL_KHR_image_pixmap
      - EGL_KHR_no_config_context
      - EGL_KHR_reusable_sync
      - EGL_KHR_surfaceless_context
      - EGL_KHR_wait_sync
      - EGL_MESA_configless_context
      - EGL_MESA_drm_image
      - EGL_MESA_image_dma_buf_export
      - EGL_NOK_texture_from_pixmap
      - EGL_WL_bind_wayland_display
  gles2:
    vendor: Intel Open Source Technology Center
    vendor: OpenGL ES-CM 1.1 Mesa 17.1.6
    extensions:
      - GL_EXT_blend_minmax
      - GL_EXT_multi_draw_arrays
      - GL_EXT_texture_filter_anisotropic
      - GL_EXT_texture_lod_bias
      - GL_OES_byte_coordinates
      - GL_OES_fixed_point
      - GL_OES_stencil_wrap
      - GL_OES_compressed_paletted_texture
      - GL_OES_query_matrix
      - GL_OES_read_format
      - GL_OES_single_precision
      - GL_EXT_texture_compression_dxt1
      - GL_OES_draw_texture
      - GL_OES_point_size_array
      - GL_OES_point_sprite
      - GL_EXT_texture_format_BGRA8888
      - GL_OES_compressed_ETC1_RGB8_texture
      - GL_OES_depth24
      - GL_OES_element_index_uint
      - GL_OES_fbo_render_mipmap
      - GL_OES_framebuffer_object
      - GL_OES_mapbuffer
      - GL_OES_rgb8_rgba8
      - GL_OES_stencil8
      - GL_OES_texture_env_crossbar
      - GL_OES_texture_mirrored_repeat
      - GL_OES_texture_npot
      - GL_OES_EGL_image
      - GL_OES_packed_depth_stencil
      - GL_OES_texture_cube_map
      - GL_APPLE_texture_max_level
      - GL_EXT_discard_framebuffer
      - GL_EXT_read_format_bgra
      - GL_OES_blend_equation_separate
      - GL_OES_blend_func_separate
      - GL_OES_blend_subtract
      - GL_OES_EGL_image_external
      - GL_OES_EGL_sync
      - GL_OES_vertex_array_object
      - GL_ANGLE_texture_compression_dxt3
      - GL_ANGLE_texture_compression_dxt5
      - GL_EXT_map_buffer_range
      - GL_KHR_debug
      - GL_OES_surfaceless_context
      - GL_EXT_compressed_ETC1_RGB8_sub_texture
      - GL_EXT_polygon_offset_clamp

**Comment by morphis** [comment]
> I also have this issue. For me systemctl --user stop anbox-session-manager.service made it work again.

Can you reliable reproduce the not happening suspend when you start the session-manager again?

**Comment by Null-Senpai** [comment]
> Can you reliable reproduce the not happening suspend when you start the session-manager again?
Huh, surprisingly not. Only happens after being logged in and having systemd start the service (because it is enabled). Stopping and then starting it made it go away for me. Strange..

**Comment by DroWnThePoor** [comment]
I know this may not be the preferred way of stopping the program, but sudo killall anbox seems to solve the problem for me.

On Thu, Aug 10, 2017 at 12:49 PM, null <notifications@github.com> wrote:

Can you reliable reproduce the not happening suspend when you start the session-manager again?
Huh, surprisingly not. Only happens after being logged in and having systemd start the service (because it is enabled). Stopping and then starting it made it go away for me. Strange..

**Comment by mwild1** [comment]
I also have this issue. Took a long and frustrating time to track it down! Any debug info I can provide?

**Comment by hpk42** [comment]
also have this issue. Seems "sudo killall anbox" allows the laptop to properly suspend. This is ubuntu 16.04 with the oneline "snap" install of anbox.

**Comment by paulmueller** [comment]
same issue with Ubuntu 16.04.3 LTS on Intel i7-4600U CPU. After 1-2 seconds the system wakes up.
This issue only appears when Anbox has been started before using the launcher. Quitting Anbox does not solve the issue.

The command ``sudo service snap.anbox.container-manager stop`` freezes the system (with fans starting to spin at full speed).

anbox system-info:
```
version: 3
snap-revision: 67
os:
  name: Ubuntu
  version: 16.04.3 LTS (Xenial Xerus)
  snap-based: true
kernel:
  version: Linux version 4.4.0-112-generic (buildd@lgw01-amd64-010) (gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.5) ) #135-Ubuntu SMP Fri Jan 19 11:48:36 UTC 2018
  binder: true
  ashmem: true
graphics:
  egl:
    vendor: Mesa Project
    version: 1.4 (DRI2)
    extensions:
      - EGL_CHROMIUM_sync_control
      - EGL_EXT_create_context_robustness
      - EGL_EXT_image_dma_buf_import
      - EGL_EXT_image_dma_buf_import_modifiers
      - EGL_KHR_config_attribs
      - EGL_KHR_create_context
      - EGL_KHR_create_context_no_error
      - EGL_KHR_fence_sync
      - EGL_KHR_get_all_proc_addresses
      - EGL_KHR_gl_renderbuffer_image
      - EGL_KHR_gl_texture_2D_image
      - EGL_KHR_gl_texture_3D_image
      - EGL_KHR_gl_texture_cubemap_image
      - EGL_KHR_image
      - EGL_KHR_image_base
      - EGL_KHR_image_pixmap
      - EGL_KHR_no_config_context
      - EGL_KHR_reusable_sync
      - EGL_KHR_surfaceless_context
      - EGL_KHR_wait_sync
      - EGL_MESA_configless_context
      - EGL_MESA_drm_image
      - EGL_MESA_image_dma_buf_export
      - EGL_NOK_swap_region
      - EGL_NOK_texture_from_pixmap
      - EGL_NV_post_sub_buffer
      - EGL_WL_bind_wayland_display
  gles2:
    vendor: Intel Open Source Technology Center
    vendor: OpenGL ES-CM 1.1 Mesa 17.2.4
    extensions:
      - GL_EXT_blend_minmax
      - GL_EXT_multi_draw_arrays
      - GL_EXT_texture_filter_anisotropic
      - GL_EXT_texture_lod_bias
      - GL_OES_byte_coordinates
      - GL_OES_fixed_point
      - GL_OES_stencil_wrap
      - GL_OES_compressed_paletted_texture
      - GL_OES_query_matrix
      - GL_OES_read_format
      - GL_OES_single_precision
      - GL_EXT_texture_compression_dxt1
      - GL_OES_draw_texture
      - GL_OES_point_size_array
      - GL_OES_point_sprite
      - GL_EXT_texture_format_BGRA8888
      - GL_OES_compressed_ETC1_RGB8_texture
      - GL_OES_depth24
      - GL_OES_element_index_uint
      - GL_OES_fbo_render_mipmap
      - GL_OES_framebuffer_object
      - GL_OES_mapbuffer
      - GL_OES_rgb8_rgba8
      - GL_OES_stencil8
      - GL_OES_texture_env_crossbar
      - GL_OES_texture_mirrored_repeat
      - GL_OES_texture_npot
      - GL_OES_EGL_image
      - GL_OES_packed_depth_stencil
      - GL_OES_texture_cube_map
      - GL_APPLE_texture_max_level
      - GL_EXT_discard_framebuffer
      - GL_EXT_read_format_bgra
      - GL_OES_blend_equation_separate
      - GL_OES_blend_func_separate
      - GL_OES_blend_subtract
      - GL_OES_EGL_image_external
      - GL_OES_EGL_sync
      - GL_OES_vertex_array_object
      - GL_ANGLE_texture_compression_dxt3
      - GL_ANGLE_texture_compression_dxt5
      - GL_EXT_map_buffer_range
      - GL_KHR_debug
      - GL_OES_surfaceless_context
      - GL_EXT_compressed_ETC1_RGB8_sub_texture
      - GL_EXT_polygon_offset_clamp
      - GL_KHR_no_error
```

**Comment by null-von-sushi** [comment]
would it somehow be possible to write a systemd rule thing to automatically kill the anbox service?

**Comment by kode54** [comment]
It appears to be triggering this event regularly:

```
Aug 07 22:54:31 dorktower dbus-daemon[8808]: apparmor="ALLOWED" operation="dbus_method_call"  bus="session" path="/org/gnome/ScreenSaver" interface="org.gnome.ScreenSaver" member="SimulateUserActivity" mask="send" name="org.gnome.ScreenSaver" pid=27881 label="snap.anbox.appmgr" peer_pid=8965 peer_label="unconfined"
```

So maybe terminate the snap container manually?

**Comment by morphis** [comment]
@kode54 Oh interesting. This seems to come from https://github.com/SDL-mirror/SDL/blob/e465610bd49c983183ad1419895f5e8108a68ae2/src/core/linux/SDL_dbus.c#L299

https://wiki.libsdl.org/SDL_EnableScreenSaver seems to be what we need to call to prevent this from happening.

**Comment by sim590** [comment]
Is it supposed to be fixed by #852? I'm using version `0.0~git20181210-1` from debian. I don't know what specific upstream version it is pointing to, but I guess it's later than october 8th and I still experience the issue.
